File: lecture004/arraybasic.cpp
#include <iostream>
#include <vector>

using namespace std;

int min(vector<int> arr)
{ //public static int min (for java)
    int min = arr[0];

    for (int i = 0; i < arr.size(); i++)
    {
        if (arr[i] < min)
        {
            min = arr[i];
        }
    }
    return min;
}
int max(vector<int> &arr)
{ //public static int max (for java)
    int max = arr[0];
    for (int i = 0; i < arr.size(); i++)
    {
        if (arr[i] > max)
        {
            max = arr[i];
        }
    }
    return max;
}
bool find(vector<int> &arr, int data)
{ //public static bool find (for java)
    //cout << "enter a no from the array to find : ";
    for (int i = 0; i < arr.size(); i++)
    { //arr.length(for java)
        if (arr[i] == data)
            //cout<<"element "<<i<<" is present";
            return true;
    }
    return false;
}
void display(vector<int> &arr)
{
    cout << "entered array is : ";
    for (int i = 0; i < arr.size(); i++)
        cout << arr[i] << " ";
}

void input(vector<int> &arr)
{
    cout << "Enter array : ";
    for (int i = 0; i < arr.size(); i++)
        cin >> arr[i];
}

int main(int args, char **argv)
{
    int a, n;
    cout << "how many elements in the array : ";
    cin >> a;
    vector<int> arr(a, 0);

    input(arr);
    display(arr);
    cout << endl;
    cout << "the minimum no in array is : " << min(arr) << endl;
    cout << "the maximum no in array is : " << max(arr) << endl;
    cout << "Enter a element to search : ";
    cin >> n;
    bool x = find(arr, n);
    if (x == 1)
    {
        cout << "elemnt is present";
    }
    else
    {
        cout << "element not present";
    }
    return 0;
}

File: lecture004/arraybasic.java
import java.util.Scanner;

public class arraybasic {
    public static Scanner scn = new Scanner(System.in); // to make it a global variable

    public static void main(String[] args) {
        System.out.print("Enter size of array : ");
        int k = scn.nextInt();
        int[] arr = new int[k];
        System.out.print("Enter array : ");
        input(arr);
        System.out.println();
        System.out.print("Entered array is : ");
        display(arr);
        System.out.println();
        System.out.println("minimum is : " + minimum(arr));
        System.out.println("maximum is : " + maximum(arr));
        int n;
        System.out.print("Enter a no to find : ");
        n = scn.nextInt();
        System.out.print(find(arr, n));
    }

    public static void display(int[] arr) {
        for (int i = 0; i < arr.length; i++)
            System.out.print(arr[i] + " ");
    }

    public static void input(int[] arr) {
        for (int i = 0; i < arr.length; i++)
            arr[i] = scn.nextInt();
    }

    public static int maximum(int[] arr) {
        int maximum;
        maximum = arr[0];
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] > maximum)
                maximum = arr[i];
        }
        return maximum;
    }

    public static int minimum(int[] arr) {
        int minimum;
        minimum = arr[0];
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] < minimum)
                minimum = arr[i];
        }
        return minimum;
    }

    public static boolean find(int[] arr, int data) {
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == data) {
                return true;
            }
        }
        return false;
    }

}